### DISCUSSION TITLE
People talking about us on the internet

### DISCUSSION THREAD

**Discussion by C4ptainCrunch** [ics-py/ics-py#315] [Show and tell]
I came across an article about `ics-py` by chance so i Googled a bit more and found a second one.
If you find some more, feel free to link them here !
* http://www.ehfeng.com/ics-vs-icalendar-python/
* https://www.nicholasnadeau.com/post/2020/6/download-ical-calendar-data-using-python/

It seems like `ics-py` is used in some nice projects too ! (One day, when we release v0.8 we should maybe help them migrate to the new API)
* https://github.com/aceisace/Inkycal
* https://github.com/simonw/datasette-ics
* https://github.com/aclu-national/elections-api
* https://github.com/wuningxi/Timetrack
* https://github.com/mobeigi/fb2cal
* https://github.com/Mini-Conf/Mini-Conf
* https://github.com/City-Bureau/city-scrapers

And also by Apache [Solr](https://github.com/apache/solr/blob/ca08f3797cc729d5aff9765d9e254ed1fb5201fb/dev-tools/scripts/requirements.txt#L5) and [Lucene](https://github.com/apache/lucene/blob/d5d6dc079395c47cd6d12dcce3bcfdd2c7d9dc63/dev-tools/scripts/requirements.txt#L5) - go figure 🤷‍♂️

**Comment by introt** [comment]
Hi! Popped in to check the status on v0.8, and decided to scroll through some of these links in case someone has already solved the problem I'm pondering writing a tool for[^tool] - though what I meant to discuss here is not related to technology, but people.

In the linked blog post, Eric Feng writes, emphasis mine:

> Unfortunately, the maintainer of ics **stopped** all open source contribution as of July this year

Regardless of when the post was written[^irony], I find the word choice troubling: aside from the maintainer themself declaring the project dead or similar[^alabaster], who's to say what's going on in the open source developer's life and whether they are to be seen again?

Quoting @N-Coder from #317 (titled "Sponsorship"):

> I started working on this project for fun while still being a student and now having transitioned to working full time since late 2019 has fastly changed the amount of free time I have at hand (and I guess the same is true for C4ptainCrunch). [...]  Even worse, I guess getting money for this project would make me feel obliged to work more on it without actually freeing up any of my time, meaning either more stress or feeling bad not doing anything in return.

I haven't been doing collaborative open source software development for long, but in that time it has dawned on me that sometimes it's not just "the end-users" that fail to see us as individuals with lives separate from the projects we volunteer for - sometimes it's our fellow devs. I don't mean to shame any individuals - I've been guilty of this as well - but in general I do feel like that the human side of open source development often lacks the appropriate consideration, resulting in the "de-individualization" of (hobbyist) developers.

Perhaps tangentially, I find the offering of money to be one manifestation of this phenomenon. As N-Coder wrote above, free time is our limiting factor, and in most cases can't be substantially increased using the amount of money offered.

I feel like this is a part of some bigger socioeconomic picture, but sadly I don't have the time to explore it further ;)

[^tool]: basically something akin to [Apple's Data Detectors](https://appleinsider.com/articles/14/04/10/how-samsung-google-teamed-up-to-steal-apple-data-detectors-for-android) for dates/events that works in batches; a "txt2ics" kinda thing. It's weird that there isn't a tool by that name already: [text2ics](https://github.com/cpg314/scripts) is a dead link, [txt2icssed](https://github.com/azizcodes/azizcodes.github.io/blob/845235a76dec7380a5dd0730e0c64991fc645113/_posts/2018-01-10-txt2icssed.md) uses sed and requires strict formatting, and [txt2ics.js](https://github.com/jeansordes/edt-miage/blob/4673eec3c2c6a2a2b3add34611f78776cd58e5c4/src/ut3/txt2ics.js) is written in a language I'm not too familiar with (pun intended; it's in French); wuningxi/Timetrack, simonw/datasette-ics and aceisace/Inkycal aren't exactly what I'm looking for either, but they're all closely related to other time management and/or visualization stuff I've also been meaning to tinker with - thanks for introducing me to them! I'll also have a look at google/python-fire for quickly implementing CLIs for some scripts, mentioned in Nicholas Nadeau's linked post.

[^irony]: It's quite ironic that a blog post discussing calendar software includes no mention of the publication date; I'd guess 2018 based on [ics2db](https://github.com/ehfeng/ics2db)'s latest commit (the blog is probs hosted on GH Pages, but that one was on their profile page). In any case, @C4ptainCrunch has demonstrably been around since.

[^alabaster]: or "undead", like what's happened with [Alabaster](https://github.com/bitprophet/alabaster/commit/6a0a56ffe79bd039085e1ad7ed26da5b217bd4ab)

**Comment by C4ptainCrunch** [comment]
I've seen https://github.com/allenporter/ical pop recently in my internet feeds. I did not have the time to read the code yet but it mentions ics-py as an implementation used as reference, quite nice :)

**Reply by allenporter** [reply]
> I've seen https://github.com/allenporter/ical pop recently in my internet feeds. I did not have the time to read the code yet but it mentions ics-py as an implementation used as reference, quite nice :)

Yes, I ended up looking to extend ics-py for home assistant use cases but ended up with some slightly different design goals so started a new project. The timeline and timespan APIs definitely straight modeled after ics-py.
